app/src/main/java/com/example/development/sakaiclientandroid/utils/DataHandler.java

```diff
@@ -58,6 +58,11 @@ public static boolean gradesRequestedForAllSites() {
         return hasRequestedAllGrades;
     }
 
```

Comment:
This technically does what is needed, but as I said the DataHandler's behavior should not be the concern of the client. Here's what I meant specifically, it probably wasn't clear previously:

```
public static void requestAllGrades(final RequestCallback UICallback, boolean refresh) {
 
    if(hasRequestedAllGrades && !refresh) {
             UICallback.onAllGradesSuccess(coursesSortedByTerm);
             return;
    }

    // Make the request here as usual....

}
```

This way, you ca specify when you want to actually refresh the data without the client having to explicitly tell DataHandler how to behave. Otherwise we'd have to make clearCache methods for every type of request and keep track of it all, which is more error prone.